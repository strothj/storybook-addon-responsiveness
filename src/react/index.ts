export { default as withResponsiveness } from "./withResponsiveness";
