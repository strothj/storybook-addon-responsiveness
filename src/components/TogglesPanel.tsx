import * as React from "react";
import { style } from "typestyle";
import { fontStyle } from "../css";
import { TogglesApi } from "./Toggles";

export interface TogglesPanelProps extends TogglesApi {}

class TogglesPanel extends React.Component<TogglesPanelProps> {
  handleEnableChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.props.setIsEnabled(event.target.checked);
  };

  handleSelectedDeviceChange = (
    event: React.ChangeEvent<HTMLSelectElement>,
  ) => {
    this.props.setSelectedDevice(event.target.value);
  };

  handleOrientationChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.props.setOrientation(event.target.checked ? "vertical" : "horizontal");
  };

  render() {
    return (
      <div className={togglesPanelStyle}>
        <div>Enable</div>
        <div>
          <input
            type="checkbox"
            checked={this.props.isEnabled}
            onChange={this.handleEnableChange}
          />
        </div>

        <div>Device</div>
        <div>
          <select
            value={this.props.selectedDevice}
            onChange={this.handleSelectedDeviceChange}
          >
            {this.props.deviceList.map(d => (
              <option key={d} value={d}>
                {d}
              </option>
            ))}
          </select>
        </div>

        <div>Vertical</div>
        <div>
          <input
            type="checkbox"
            checked={this.props.orientation === "vertical"}
            onChange={this.handleOrientationChange}
          />
        </div>
      </div>
    );
  }
}

const togglesPanelStyle = style(
  {
    boxSizing: "border-box",
    display: "flex",
    flexWrap: "wrap",
    alignContent: "flex-start",
    width: "100%",
    padding: 5,

    $nest: {
      "&> *": {
        boxSizing: "border-box",
      },

      "div:nth-child(odd)": {
        width: 100,
        textAlign: "right",
        textTransform: "uppercase",
      },

      "div:nth-child(even)": {
        width: "calc(100% - 100px)",
      },

      div: {
        padding: 8,
      },
    },
  },
  fontStyle,
);

export default TogglesPanel;
