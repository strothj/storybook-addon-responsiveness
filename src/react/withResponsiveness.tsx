import { StoryDecorator } from "@storybook/react";
import addons from "../addons";
import { eventEnableStoryResponsiveness } from "../constants";

const withResponsiveness: StoryDecorator = (render, { kind, story }) => {
  const channel = addons.getChannel();

  channel.emit(eventEnableStoryResponsiveness, { kind, story });

  return render() as any;
};

export default withResponsiveness;
