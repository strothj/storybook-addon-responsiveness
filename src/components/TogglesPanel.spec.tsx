import * as React from "react";
import * as renderer from "react-test-renderer";
import TogglesPanel from "./TogglesPanel";

jest.mock("../settings");

it("renders toggle switches", () => {
  const component = renderer.create(
    <TogglesPanel
      deviceList={["iPhone 5/SE"]}
      isEnabled
      orientation="vertical"
      selectedDevice="iPhone 5/SE"
      setIsEnabled={jest.fn()}
      setOrientation={jest.fn()}
      setSelectedDevice={jest.fn()}
    />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
