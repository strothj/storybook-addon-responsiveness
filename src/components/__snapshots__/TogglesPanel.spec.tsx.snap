// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toggle switches 1`] = `
<div
  className="f252qlb"
>
  <div>
    Enable
  </div>
  <div>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div>
    Device
  </div>
  <div>
    <select
      onChange={[Function]}
      value="iPhone 5/SE"
    >
      <option
        value="iPhone 5/SE"
      >
        iPhone 5/SE
      </option>
    </select>
  </div>
  <div>
    Vertical
  </div>
  <div>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
